importer/api/__init__.py

```diff
@@ -109,7 +108,8 @@ def import_course_from_path(path, repo_id, user_id):
     )
     bundle.import_from_directory(path)
     static_dir = join(path, 'static')
-    course = import_course(bundle, repo_id, user_id, static_dir)
+    with transaction.atomic():
+        course = import_course(bundle, repo_id, user_id, static_dir)
```

Comment:
this can be a lot of stuff in a single transaction... what the impact on the memory for the DB?